elastalert/elastalert.py

```diff
@@ -361,6 +388,31 @@ def get_hits_terms(self, rule, starttime, endtime, index, key, qk=None, size=Non
         elastalert_logger.info('Queried rule %s from %s to %s: %s buckets' % (rule['name'], pretty_ts(starttime, lt), pretty_ts(endtime, lt), len(buckets)))
         return {endtime: buckets}
 
+    def get_hits_aggregation(self, rule, starttime, endtime, index, query_key, term_size=None):
+        rule_filter = copy.copy(rule['filter'])
+        base_query = self.get_query(rule_filter, starttime, endtime, timestamp_field=rule['timestamp_field'], sort=False, to_ts_func=rule['dt_to_ts'], five=self.is_five())
+        if term_size is None:
+            term_size = rule.get('terms_size', 50)
+        query = self.get_aggregation_query(base_query, rule, query_key, term_size)
+        try:
+            if not self.is_five():
+                res = self.current_es.search(index=index, doc_type=rule.get('doc_type'), body=query, search_type='count', ignore_unavailable=True)
+            else:
+                res = self.current_es.search(index=index, doc_type=rule.get('doc_type'), body=query, size=0, ignore_unavailable=True)
+        except ElasticsearchException as e:
+            if len(str(e)) > 1024:
```

Comment:
This is copy/pasta'd in 4 places now. Maybe make this a util function or something?